website/docs/reference/resource-configs/snowflake-configs.md

```diff
@@ -9,6 +9,393 @@ To-do:
+
+<File name='dbt_project.yml'>
+
+```yaml
+
+flags:
+  enable_iceberg_materializations: True
+
+```
+
+</File>
+
+
+The following configurations are supported: 
+
+| Field | Type | Required | Description | Sample input |
+|-------|------|----------|-------------|-------|
+|Table Format | String | Yes | Configures the object table format. | `iceberg` |
+|External volume | String | Yes | Specifies the identifier (name) of the external volume where the Iceberg table stores its metadata files and data in Parquet format. If you don’t specify this parameter, the Iceberg table defaults to the external volume for the schema, database, or account. The schema takes precedence over the database, and the database takes precedence over the account.| `my_s3_bucket` |
+|Base location Subpath |String | No | Defines the directory path for Snowflake to write the table data and metadata files. If you change this after creating the table, it will write a new table but leave the existing one. | `jaffle_marketing_folder` |
+
+### Example configuration
+
+To configure an Iceberg table materialization in dbt, refer to the example configuration:
+
+<File name='models/<modelname>.sql'>
+
+```sql
+
+{{
+  config(
+    materialized = "table",
+    table_format="iceberg",
+    external_volume="s3_iceberg_snow",
+    base_location_subpath="jaffle_shop_folder",
+  )
+}}
+
+select * from {{ ref('raw_orders') }}
+
+```
+
+</File>
+
+### Base location 
+
+If you were to create an Iceberg table in the Snowflake query editor, you would be required to provide a `base_location` as an input. dbt abstracts away from that. The default behavior in dbt is to provide a `base_location` string that follows this convention:
+
+`_dbt/{SCHEMA_NAME}/{MODEL_NAME} `
+
+However, dbt allows users to configure a `base_location_subpath` that is empty by default but, when provided, will be concatenated to the end of the previously described pattern for `base_location` string generation. We recommend leaving the `base_location_subpath` field blank by default. The `base_location_subpath` input is always appended to your schema name, and this behavior cannot be overridden. This ensures dbt can differentiate Iceberg model builds by environment.
+
+dbt does this to enforce best practices. With Snowflake-managed Iceberg format tables, the user owns and maintains the data storage of the tables in an external storage solution (the declared `external volume`). The base location declares where to write in the external volume. The Snowflake Iceberg catalog will keep track of your Iceberg table regardless of where the data lives within the `external volume` declared and what `base_location` is provided. However, passing anything into the `base_location` field, including an empty string, is possible.  If a project does not enforce an input and always allows empty strings, this could result in future technical debt because it will limit the ability to:
```

Comment:
#### Why use the dbt generated `base_location`?

dbt manages `base_location` on behalf of users to enforce best practices. With Snowflake-managed Iceberg format tables, the user owns and maintains the data storage of the tables in an external storage solution (the declared `external volume`). The `base_ location` parameter declares where to write the data within the external volume. The Snowflake Iceberg catalog keeps track of your Iceberg table regardless of where the data lives within the `external volume` declared and the `base_location` provided. However, Snowflake permits passing anything into the `base_location` field, including an empty string, even reusing the same path across multiple tables. This behavior could result in future technical debt because it will limit the ability to: